### ISSUE TITLE
Can't set permissions for computed field when function lives in schema different from table

### ISSUE THREAD

**Issue by strazan** [hasura/graphql-engine#7536]
### Version Information

Server Version: v2.0.9

### Environment

macos

### What is the expected behaviour?

When adding a scalar computed fields in a table I expect to be able to change the permissions for that column under the permissions tab.

### What is the current behaviour?

Currently this works just fine if the function lives under the same schema as the table adding it as a computed field.
But when **adding a computed field** with a function **under a different schema**, the column permission checkbox doesn't exist. 

One can go to api explorer and query the computed field to see that it actually works. By having `x-hasura-role` **not** enabled, one received data. 
### How to reproduce the issue?

1. Create a table
 ```SQL
CREATE TABLE IF NOT EXISTS export.calculator
(
    value_one numeric,
    value_two numeric
);
```
2. Create a function in a different schema
```SQL
CREATE OR REPLACE FUNCTION public.sum(cal_row export.calculator)
   RETURNS numeric
   LANGUAGE sql
   STABLE
AS $function$
BEGIN
   RETURN cal_row.value_one + cal_row.value_two;
END;
$function$
```
3. In Hasura add the function as a computed field to the table.
4. Navigate to the permissions tab and notice how you can't change column permissions for the computed field.

### Any possible solutions?

This might not actually be a bug, if it's not, I'd suggest better documentation of this case. Maybe even a tooltip for `Function schema`.
![image](https://user-images.githubusercontent.com/43888637/132689275-15d29b4c-4b50-4580-8ed8-12fe513237c6.png)

**Comment by vijayprasanna13** [comment]
fixed in commit 569fb3d21780c4bb9200f21f04c9178becd4abda

**Comment by strazan** [comment]
:D
